### PR TITLE
sdk: Fix nonce conflict issues with concurrent transactions

### LINKED ISSUE #1952: Transaction nonce is overrun when using Wallet
This is an old well known issue from Raiden core(e.g. https://github.com/raiden-network/raiden/issues/804), and also have shown up on `ethers` repo multiple times (e.g. https://github.com/ethers-io/ethers.js/issues/435). `ethers` even introduced its own [NonceManager](https://github.com/ethers-io/ethers.js/blob/master/packages/experimental/src.ts/nonce-manager.ts) to deal with the issue.

This happens on ethers and in our codebase when using `Wallet` signer because of how it fills transaction requests missing an explicit nonce (which are all of our current transactions): it makes an **async** network request to get provider's **view** of our `transactionCount`, and races here can happen both on the async network request and also even on the same ETH node, transactionCount can lag behind after a transaction gets submitted and acknowledged on the pool.

This show up **mainly** when multiple transactions are being sent simultaneously/in parallel, which is more common on the Scenario Player. On Metamask (even using subkey), this doesn't seem to be common, since looks like MM keeps a local view of this count, but is on CLI.

One approach, taken by Core and ether's NonceManager, is to keep a local nonce counter, and use it on new transactions. Drawback is that it can easily get out of sync if transactions are performed outside of the application (which is often the case with Metamask accounts), and also if a transaction errors for some reason and isn't broadcast/mined (which we use/relay a lot to save gas), you end up with a "hole" in the transaction pool, which won't get new transactions through before you somehow can fill it. Several instances of both issues can be seen on the mentioned repos.

An alternative approach (which I'd prefer) is to retry (re-subscribe) the observables which contain transaction sending operations and validation. The advantages include that we can have strict control over from where we're retrying (e.g. `approve` + `deposit`, where the `approve` is only called if needed, and we can check `allowance` on each retry). by relying on the already existing mechanism which depends on `getTransactionCount` and `estimateGas` to test if transactions would succeed before actually sending them, we can make sure the above issues never fail. The drawback is a delay while retrying before giving up on transactions which won't succeed, but since on-chain transactions don't happen that often, 40s-1min of retries for txs which won't suceed is a small price for reliability here. Retried network requests/wasted tx signatures are cheaper than reverted transactions or "hole"-like nonce sync loss.

This was identified on the `BF1` scenario (#1948).

## Steps to Reproduce
1. Using CLI, have multiple `openChannel` calls be done to different nodes simultaneously, with a `deposit` set

## Expected Result
- They eventually succeed

## Actual Result
- One of the openChannel calls will fail, and probably both deposits

## Additional Infos (Optional) 
```
- Browser & version
- Browser console errors
- Device / OS
```

### PR DESCRIPTION
**Pull request by andrevmatos**
Fixes #1952 

**Short description**
Implements the alternative approach described in the issue.
BF6 scenario pass fully with it.
BF1 passes the channel opening step which was failing and was base for the creation of the issue.

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Test scenarios

### CODE REVIEW

File: raiden-ts/src/channels/utils.ts [hunk 1]
```diff
@@ -165,31 +173,71 @@ export function assertTx(
...
-        return receipt as ContractReceipt & { transactionHash: Hash; blockNumber: number };
+        return [tx, receipt] as [
+          ContractTransaction,
+          ContractReceipt & { transactionHash: Hash; blockNumber: number },
+        ];
       }),
     );
 }
 
+export const txNonceErrors: readonly string[] = [
```
Comment (weilbith) [inline raiden-ts/src/channels/utils.ts:198]: Where did you took these errors from? No chance to get them actually as "code" from EthersJS? :thinking:

File: raiden-ts/src/channels/epics.ts [hunk 2]
```diff
@@ -914,18 +918,22 @@ export const channelCloseEpic = (
...
+              address,
+              balanceHash,
+              nonce,
+              additionalHash,
+              nonClosingSignature,
+              closingSignature,
+            ),
+          ).pipe(
+            assertTx('closeChannel', ErrorCodes.CNL_CLOSECHANNEL_FAILED, { log }),
+            retryTx(provider.pollingInterval, undefined, undefined, { log }),
```
[the remark this answers is not in the document]
Comment (andrevmatos) [reply raiden-ts/src/channels/epics.ts:934]: There's nothing in parallel at sight, but we need to cover all contracts calls with it, as these could happen in parallel with some other request: e.g. a script request depositing in a channel at the same time it tries to close another

File: raiden-ts/src/channels/utils.ts (hunk 1 above)
Comment (andrevmatos) [reply raiden-ts/src/channels/utils.ts:198]: These varies from eth node being used (parity, geth, infura?). These codes I got in practice, by triggering the bug several times with the scenarios.

File: raiden-ts/src/channels/epics.ts [hunk 3]
```diff
@@ -620,15 +628,9 @@ const makeDeposit$ = (
...
-  approveNonce?: number,
 ): Observable<channelDeposit.failure> => {
   if (!deposit?.gt(Zero)) return EMPTY;
-
-  function stopRetryIfRejected(err: { message: string; code?: string | number }, count: number) {
-    // 4001 is Metamask's code for rejected/denied signature prompt
-    const metamaskRejectedErrorCode = 4001;
-    return count >= 10 || err?.code === metamaskRejectedErrorCode;
-  }
+  const pollingInterval = (tokenNetworkContract.provider as JsonRpcProvider).pollingInterval;
```
[the remark this answers is not in the document]
Comment (andrevmatos) [reply raiden-ts/src/channels/epics.ts:633]: The problem is that the whole batch below it is a single pipe chain, so we can't declare anything in the meantime, but I can remove the declared variable and use it directly in the only place it's used.

**Pull request merged**